### DISCUSSION TITLE
Bundle CSS module styles into single compiled file? (Parcel 2, NextJS)

### DISCUSSION THREAD

**Discussion by dowrow** [parcel-bundler/parcel#5939] [Q&A] [answered]
I'm using parcel to build each component of my mono-repo.
Every package has a package.json similar to:
```
{
  "name": "@myscope/accordion",
  "version": "1.0.7",
  "repository": "https://github.com/myscope/myrepo",
  "description": "Display a collapsed resume that expands on click",
  "main": "dist/main/index.js",
  "module": "dist/module/index.js",
  "license": "MIT",
  "scripts": {
    "build": "parcel build Accordion.js"
  },
  "dependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  }
}
```

I'm using the following deps on the root package.json of the mono-repo:
```
    "autoprefixer": "^10.2.4",
    "parcel": "^2.0.0-beta.1",
    "postcss": "^8.2.6",
    "postcss-modules": "^3.0.0",
```
because components use CSS modules.

They build just fine. However, when I publish them and try to use them on a NextJS project I get:
```
ready - started server on http://localhost:3000
info  - Using external babel configuration from /...omitted.../.babelrc
error - ./node_modules/@scope/accordion/dist/module/index.css
Global CSS cannot be imported from within node_modules.
Read more: https://err.sh/next.js/css-npm
```

Is there any way to "compile" the package or bundle/concat the css module into the output file with Parcel 2?
I'm can't find any docs about that.

**Comment by devongovett** [comment, marked as answer]
This is because Next.js doesn't support loading CSS files from within node_modules. We have the same problem in our component library. I guess overriding the Next.js webpack config is the only way. Let me know if you figure it out!

**Reply by dowrow** [reply]
Thank you for the answer, @devongovett !
Yup, looks like this is a common pain with Next.
For anyone interested [this thread](https://github.com/vercel/next.js/issues/12079) may be valuable.
